sourcerer-dbstore/src/main/java/org/elder/sourcerer2/dbstore/DbstoreEventRepository.kt

```diff
@@ -0,0 +1,254 @@
+package org.elder.sourcerer2.dbstore
+
+import com.fasterxml.jackson.core.type.TypeReference
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.google.common.collect.ImmutableList
+import com.google.common.collect.ImmutableMap
+import org.elder.sourcerer2.EventData
+import org.elder.sourcerer2.EventRecord
+import org.elder.sourcerer2.EventRepository
+import org.elder.sourcerer2.EventSubscriptionUpdate
+import org.elder.sourcerer2.ExpectedVersion
+import org.elder.sourcerer2.RepositoryReadResult
+import org.elder.sourcerer2.RepositoryVersion
+import org.elder.sourcerer2.StreamId
+import org.elder.sourcerer2.StreamReadResult
+import org.elder.sourcerer2.StreamVersion
+import org.reactivestreams.Publisher
+import org.slf4j.LoggerFactory
+
+internal class DbstoreEventRepository<T>(
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Classes with many functions tend to do too many things and often come in conjunction with large classes and can quickly become God classes. Consider extracting methods to (new) classes better matching their responsibility. SIZE 17 is greater than the threshold 10. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.elder.org/coding_rules#rule_key=detekt-kotlin%3ATooManyFunctions)